src/main/java/net/minecraftforge/common/util/RayTraceUtils.java

```diff
@@ -0,0 +1,59 @@
+package net.minecraftforge.common.util;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+
+public final class RayTraceUtils
+{
```

Comment:
Raytracing from the player's point of view without depending on client-side-specific code.